src/arrayMethodJoin.js

```diff
@@ -4,8 +4,19 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const newArray = [];
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      this[i] === null || this[i] === undefined
```

Comment:
Please try to avoid ternary operator, please try to use **if**. Also it's redundant to make newArray[i] = '' - empty string. If the condition is met, you can push this[i], if i !== this.length - 1   -  push separator. Example for first condition:
```suggestion
      if (this[i] !== null || this[i] !== undefined) {
        newArray.push(this[i])
      }
```